notions_de_base/exercice4.js

```diff
@@ -0,0 +1,23 @@
+const getTheBill = (card) => {
+    if (isEmpty(card)) throw "Your card is empty";
+    if (hasNoValidPriceOrQuantity(card)) throw "Price or quantity missing on one object";
+    let finalBill = 0;
+    card.map(object => {
```

Comment:
Pourquoi un map ? As-tu besoin de retourner un tableau ?

Reply:
Ben c'est toi qui m'avait dit d'utiliser .map comme méthode

Reply:
Tu es sure ?

As-tu bien compris la différence entre map et foreach ? Sinon n'hésite pas à relire la doc ;)

Reply:
Oui j'ai compris que map retournait un tableau contrairement à forEach. Mais j'ai utilisé map parce que tu m'avais dit que pour les exos 3 et 4 il fallait que j'utilise cette méthode. Du coup je change ?

Reply:
Ah bah j'ai dit une connerie alors ^^
Pour l'exo3 oui c'est pertinent.
Bah écoute à toi de voir, est-ce que tu as besoin d'un map ici ?

Reply:
non je pense que comme tu le dis un forEach serait plus approprié. Je m'en occupe !

Reply:
- [ ] changing .map for forEach method